src/python/tests/assembly/test_status.py

```diff
@@ -0,0 +1,343 @@
+    elif nextflow_cachedir:
+        expected_cache_path = Path(nextflow_cachedir)
+    elif singularity_cachedir:
+        expected_cache_path = Path(singularity_cachedir)
+    else:
+        expected_cache_path = Path()
+    # Get expected container URL used to pull the container
+    if datasets_version:
+        expected_container_url = datasets_version
+    else:
+        expected_container_url = DATASETS_SINGULARITY["datasets_version_url"]
+    # Patch the environment variables
+    new_env = {}
+    if nextflow_cachedir:
+        new_env["NXF_SINGULARITY_CACHEDIR"] = nextflow_cachedir
+    if singularity_cachedir:
+        new_env["SINGULARITY_CACHEDIR"] = singularity_cachedir
+    with patch.dict(os.environ, new_env, clear=True):
+        assert singularity_image_setter(sif_cache_path, datasets_version)
+    # Check that the spython pull method was called with the right arguments
+    mock_client.pull.assert_called_with(
+        expected_container_url, stream=False, pull_folder=expected_cache_path, quiet=True
+    )
+
+
+@pytest.mark.parametrize(
+    "file_name, expected_output, expectation",
+    [
+        param("assemblies.txt", ["GCA_900524668.2", "GCF_123456789.1"], does_not_raise(), id="Valid file"),
+        param("meta_report.tsv", [], raises(UnsupportedFormatError), id="Wrong accession format"),
+    ],
+)
+def test_get_assembly_accessions(
+    data_dir: Path, file_name: str, expected_output: list[str], expectation: ContextManager
+) -> None:
+    """Tests the `get_assembly_accessions()` function.
+
+    Fixtures:
+        data_dir
+
+    Args:
+        file_name: File with one line per INSDC assembly accession.
+        expected_output: Expected assembly accessions returned.
+        expectation: Context manager of expected raise exception.
+    """
+    file_path = data_dir / file_name
+    with expectation:
+        result = get_assembly_accessions(file_path)
+        assert result == expected_output
+
+
+def test_fetch_accessions_from_core_dbs(
+    request: FixtureRequest,
+    tmp_path: Path,
+    data_dir: Path,
+    db_factory: Callable[[Path | None, str | None], UnitTestDB],
+) -> None:
+    """Tests the `fetch_accessions_from_core_dbs()` function.
+
+    Fixtures: request, tmp_path, data_dir, db_factory
+    """
+    # Create all test databases and add their names into a flat file
+    test_dbs = [
+        db_factory(data_dir / "my_core_db", "my_core_db"),
+        db_factory(data_dir / "another_core_db", "another_core_db"),
+        db_factory(data_dir / "yet_another_core_db", "yet_another_core_db"),
+    ]
+    tmp_file = tmp_path / "core_dbs_list.txt"
+    with tmp_file.open("w") as fin:
+        for db in test_dbs:
+            fin.write(f"{db.dbc.db_name}\n")
+    test_server = make_url(request.config.getoption("server"))
+    accessions = fetch_accessions_from_core_dbs(tmp_file, test_server)
+    assert not DeepDiff(accessions, {f"{os.environ['USER']}_my_core_db.db": "GCF_001194135.2"})
```

Comment:
Part of this issue appears to be related to SQLAlchemy 1.4.45 vs 2.0. I was able to get all tests to pass with --server, but had to alter the deepdiff line:256....  {f"{os.environ['USER']}_my_core_db.db -> {}_my_core_db. My venv has alchemy v1.4.45 installed.